### PR TITLE
I have added three patches to run ocid as a system service

### PR DESCRIPTION
**Pull request by rhatdan**
Probably need to move systemNotify to a different place in the code to indicate that ocid is ready to receive commands from k8s.

### CODE REVIEW

**Comment by rhatdan** [comment]
Also was not sure about vendoring in the patches from coreos.  When I played with vendor.sh it destroyed my vendor subdirs.  So I manually pulled them in.

**Comment by runcom** [comment]
It's normal that the vendor scripts firstly purge the vendor directory. I believe you can just vendor dependencies one by one with hack/vendor.sh github.com/coreos/somepkg

**Comment by rhatdan** [comment]
I thought it was some magical script that went though all of the imports and grabbed what it needed.

**Comment by cyphar** [comment]
We need to make socket activations / talking to systemd optional (based on whether or not you used socket activation). One thing I'd like is if we took a cue from docker/containerd#178 and used some `proto://addr` to indicate whether to use systemd socket activation or just normal binding.

**Comment by cyphar** [comment]
Also, I've also had a lot of issues recently with `hack/vendor.sh`. It deletes things it shouldn't and adds files it shouldn't as well...

**Comment by rhatdan** [comment]
I think socket activation would need to be a separate pull.  Need to hand off the open socket.  
@mrunalp  could you check to see when the systemNotification call should happen?

**Comment by mrunalp** [comment]
@rhatdan We can call it somewhere here https://github.com/kubernetes-incubator/cri-o/blob/master/cmd/server/main.go#L190
Also, we can make this optional.

**Comment by rhatdan** [comment]
@mrunalp Not much difference from where I called it, so I would figure I can leave it where it is.  Not sure what you mean by optional?  I don't think this can fail, and will just be ignored if not run from a systemd unit file.

**Comment by cyphar** [comment]
> Not much difference from where I called it, so I would figure I can leave it where it is.

I think it is quite different. You don't want to signal systemd that we're "ready" until we're actually ready to serve requests. This is why Docker returns 404s a few seconds after starting under systemd.

**Comment by rhatdan** [comment]
Yes you are right.  I have my head up my butt this morning.

**Comment by rhatdan** [comment]
Also removed docker-storage-setup, although I think we will need something like that to configure storage, especially if we are going to use devicemapper backend.

File: hack/vendor.sh [hunk 1]
```diff
@@ -46,6 +46,7 @@ case $# in
 	;;
 esac
 
+clone git github.com/coreos/go-systemd/daemon
```
Comment (mrunalp) [inline hack/vendor.sh]: We are still missing the vendored code in this PR. could you run hack/vendor.sh for just this dependency?

Comment (runcom) [inline hack/vendor.sh]: `./hack/vendor.sh github.com/coreos/go-systemd` - the script will import the `daemon` sub-pkg correctly (hopefully)

**Pull request closed without merging**